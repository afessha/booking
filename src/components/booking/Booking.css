body {
  padding-top: 5px;
  padding-bottom: 10px;
  background-color: #eee;
}

.form-signin {
  max-width: 330px;
  padding: 10px;
  margin: 0 auto;
  text-align: center;
}
.form-signin .form-signin-heading {
  margin-bottom: 10px;
  color: white;
}
.container {
  background-color: #122f4c;
  padding: 0;
}

.form-signin label {
  color: white;
  font-size: 18px;
}
.form-signin input {
  border-radius: 2px;
}

.form-signin input {
  margin-bottom: 10px;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
}

.footer {
  margin: 10px;
  text-align: center;
  color: #122f4c;
  font-weight: bold;
}
.error {
  color: red;
}
.submitted {
  color: green;
}
