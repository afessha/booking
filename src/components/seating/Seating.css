.search-input {
  text-align: center;
}
.seating-title {
  text-align: center;
}
.link-to-booking {
  align-items: center;
}
