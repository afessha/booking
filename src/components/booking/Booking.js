import React from "react";
import { connect } from "react-redux";
import axios from "axios";
import "./Booking.css";
import Footer from "../footer/Footer";
import { useFormik } from "formik";
import * as Yup from "yup";
import { bookingAdd } from "../../redux/actions";

function Booking(props) {
  const formik = useFormik({
    initialValues: {
      firstname: "",
      lastname: "",
      dining_date: "",
      number_of_covers: "",
      booking_email: "",
      phone_number: "",
      status: "Not arrived",
    },
    validationSchema: Yup.object({
      firstname: Yup.string()
        .max(15, "Must be 15 characters or less")
        .required("Required"),
      lastname: Yup.string()
        .max(20, "Must be 20 characters or less")
        .required("Required"),
      booking_email: Yup.string()
        .email("Invalid email address")
        .required("Required"),
    }),
    onSubmit: async (booking) => {
      try {
        let res = await axios.post(
          "https://a6qroxzfnb.execute-api.eu-west-2.amazonaws.com/dev/bookings",
          JSON.stringify(booking)
        );
        console.log(res.status);
        if (res.status === 200) {
          props.addNew({ booking });
          formik.resetForm();
        }
      } catch (err) {
        console.log("Error happened" + err);
      }
    },
  });

  return (
    <>
      <div>
        <div className="container">
          <form className="form-signin" onSubmit={formik.handleSubmit}>
            <h2 className="form-signin-heading">Booking</h2>
            <br />
            <label>First Name</label>
            <input
              className="form-control"
              name="firstname"
              {...formik.getFieldProps("firstname")}
            />
            {formik.touched.firstname && formik.errors.firstname ? (
              <div className="error">{formik.errors.firstname}</div>
            ) : null}
            <label>Last Name</label>
            <input
              className="form-control"
              name="lastname"
              {...formik.getFieldProps("lastname")}
            />
            {formik.touched.lastname && formik.errors.lastname ? (
              <div className="error">{formik.errors.lastname}</div>
            ) : null}
            <label> Dining Date</label>
            <input
              className="form-control"
              name="dining_date"
              {...formik.getFieldProps("dining_date")}
            />
            <label>Number of Covers </label>
            <input
              className="form-control"
              name="number_of_covers"
              {...formik.getFieldProps("number_of_covers")}
            />

            <label> Email </label>
            <input
              className="form-control"
              name="booking_email"
              {...formik.getFieldProps("booking_email")}
            />
            {formik.touched.booking_email && formik.errors.booking_email ? (
              <div className="error">{formik.errors.booking_email}</div>
            ) : null}
            <label>Phone Number</label>
            <input
              className="form-control"
              name="phone_number"
              {...formik.getFieldProps("phone_number")}
            />
            <button className="btn btn-lg btn-success btn-block" type="submit">
              Book Table
            </button>
          </form>
        </div>
        <Footer />
      </div>
    </>
  );
}

const mapDispatchToProps = (dispatch) => {
  return {
    addNew: (booking) => dispatch(bookingAdd(booking)),
  };
};
export default connect(null, mapDispatchToProps)(Booking);
