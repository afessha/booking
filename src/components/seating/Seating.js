import React from "react";
import "./Seating.css";
import { connect } from "react-redux";
import Footer from "../footer/Footer";
import { bookingLoad } from "../../redux/actions";
function Seating({ bookings }) {
  console.log(bookings);
  return (
    <div>
      <h3 className="seating-title">Booking List </h3>
      <div className="link-to-booking"></div>
      <br />
      <br />

      <table className="table table-hover">
        <thead>
          <tr>
            <th scope="col">First Name </th>
            <th scope="col">Last Name </th>
            <th scope="col">Dining Date</th>
            <th scope="col">Number of Covers </th>
            <th scope="col">Phone Number</th>
            <th scope="col">Email</th>
            <th scope="col">Status</th>
          </tr>
        </thead>
        {bookings.map((result) => (
          <tbody>
            <tr>
              <td>{result.firstname} </td>

              <td>{result.lastname} </td>

              <td>{result.diningDate} </td>

              <td>{result.numberOfCovers} </td>

              <td>{result.phoneNumber}</td>

              <td>{result.email}</td>
              <td>
                <select>
                  <option value="notArrived">Not arrived</option>
                  <option value="seated">Seated</option>
                </select>
              </td>
            </tr>
          </tbody>
        ))}
      </table>
      <Footer />
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    bookings: state.bookings,
  };
};

export default connect(mapStateToProps)(Seating);
