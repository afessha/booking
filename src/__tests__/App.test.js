import React from "react";
import { render } from "@testing-library/react";
import { create } from "react-test-renderer";
import Booking from "../components/booking/Booking";
import Seating from "../components/seating/Seating";
import { BrowserRouter, Route } from "react-router-dom";

function App() {
  return (
    <BrowserRouter basename="/booking">
      <Route exact={true} path="/" component={Booking} />
      <Route path="/booking-list" component={Seating} />
    </BrowserRouter>
  );
}

export default App;

describe("Footer component", () => {
  test("Matches the snapshot", () => {
    const app = create(<App />);
    expect(app.toJSON()).toMatchSnapshot();
  });
});
