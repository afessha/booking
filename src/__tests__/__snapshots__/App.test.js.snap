// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Matches the snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <form
      className="form-signin"
    >
      <h2
        className="form-signin-heading"
      >
        Booking
      </h2>
      <a
        href="/booking-list"
        onClick={[Function]}
      >
         Go to Bookings List
      </a>
      <p
        className="submitted"
      >
         
        
      </p>
      <br />
      <label>
        First Name
      </label>
      <input
        className="form-control"
        name="firstname"
        onChange={[Function]}
        required={true}
      />
      <label>
        Last Name
      </label>
      <input
        className="form-control"
        name="lastname"
        onChange={[Function]}
        required={true}
      />
      <label>
         Dining Date
      </label>
      <input
        className="form-control"
        name="diningDate"
        onChange={[Function]}
        required={true}
      />
      <label>
        Number of Covers 
      </label>
      <input
        className="form-control"
        name="numberOfCovers"
        onChange={[Function]}
        required={true}
      />
      <label>
         Email 
      </label>
      <input
        className="form-control"
        name="email"
        onChange={[Function]}
        required={true}
      />
      <label>
        Phone Number
      </label>
      <input
        className="form-control"
        name="phoneNumber"
        onChange={[Function]}
        required={true}
      />
      <button
        className="btn btn-lg btn-success btn-block"
        onClick={[Function]}
        type="submit"
      >
        Book Table
      </button>
    </form>
  </div>
  <div
    className="footer"
  >
     Awet Fessha @2020 
  </div>
</div>
`;
